internal/types/document_validation_test.go

```diff
@@ -117,4 +117,40 @@ func TestDocumentValidateData(t *testing.T) {
 			}
 		})
 	}
+
+	for name, tc := range map[string]struct {
```

Comment:
Let's make the previous `for` and this `for` subtests, so we can run them in parallel.